storage_controller/src/service/safekeeper_service.rs

```diff
@@ -151,11 +151,34 @@ impl Service {
             "Got {} non-successful responses from initial creation request of total {total_result_count} responses",
             remaining.len()
         );
-        if remaining.len() >= 2 {
+        let target_sk_count = timeline_persistence.sk_set.len();
+        let quorum_size = match target_sk_count {
```

Comment:
Shouldn't this logic live in `safekeepers_for_new_timeline`? With the current position we persist in the db, which doesn't seem right if we only have two safekeepers configured.

Reply:
made it not support 1 and 2 in that function as well, although I haven't changed it here, as it might still encounter it from the db.

Reply:
> as it might still encounter it from the db.

Just for my curiosity. When would we hit such a case?

Reply:
when they started out with a `#[cfg(test)]` safekeeper before and then they added safekeepers and removed that. Then, retrying creations would get you into that situation. I know it's highly unlikely, but still better to write robust code than non-robust one IMO.